packages/@o3r/schematics/src/utility/loaders.ts

```diff
@@ -6,6 +7,34 @@ import * as path from 'node:path';
 import type { PackageJson } from 'type-fest';
 import type { WorkspaceProject, WorkspaceSchema } from '../interfaces/index';
 
+function findFilenameInTreeRec(memory: Set<FileEntry>, directory: DirEntry, fileMatchesCriteria: (file: string) => boolean, ignoreDirectories: string[]) {
+  if (ignoreDirectories.some(i => directory.path.includes(i))) {
+    return memory;
+  }
+
+  directory.subfiles
+    .filter(fileMatchesCriteria)
+    .forEach((file) => memory.add(directory.file(file)!));
+
+  directory.subdirs
+    .forEach((dir) => findFilenameInTreeRec(memory, directory.dir(dir), fileMatchesCriteria, ignoreDirectories));
```

Comment:
hopefully there is no symlinks in the tree, so no potential infinite loops ?

Reply:
I think you can control this through the `ignoreDirectories` param. Do you think it's safer to add a depth limit?

Reply:
Good point, how do symlinks are translated inside the tree object?

Reply:
I did a test locally with a symlinked package in my node_modules and the tree is flattened so there's no risk for infinite loop

Reply:
did you test with 2 symlinked folder referencing each other ?
```
node_modules
|  a
|  |  node_modules
|  |  |  b (symlink)
|  b
|  |  node_modules
|  |  |  a (symlink)
```

Reply:
No, I have not tried.
Should we ever access the `node_modules` folder? I haven't identified a use case for it. 
I propose that we always ignore the `node_modules` folders and we document this in the JSDoc. 
Not excluding `node_modules` considerably slows down this method.

Reply:
> did you test with 2 symlinked folder referencing each other ?
> 
> ```
> lib
> |  a
> |  |  node_modules
> |  |  |  b (symlink)
> |  b
> |  |  node_modules
> |  |  |  a (symlink)
> ```

Tests done with 2 folders linking one each other and we didn't run into no infinite loops.